### PR TITLE
Feature/add e2e correctness tests

### PR DESCRIPTION
**Pull request by dskwhitehat**

### CODE REVIEW

File: e2e-correctness-tests/test_ursa_query.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            with open(testdir + query_file) as f:
+                data = f.read()
+
+            logging.info("Query: " + str(data))
+            socket.send_string(data)
+
+            resp = json.loads(socket.recv_string()).get("result").get("files")
+            logging.info("Files: " + str(resp))
+
+            if (query_file[:-4] == "negative") and (len(resp) != 0):
```
Comment (msm-code) [inline e2e-correctness-tests/test_ursa_query.py:39]: maybe `query_file.startswith('negative')`?

File: e2e-correctness-tests/test_ursa_query.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                logging.info(
+                    "Test failed for "
+                    + str(query_file)
+                    + ". "
+                    + str(len(resp))
+                    + " files found: "
+                    + str(resp)
+                )
+                failures.append(query_file)
+            elif (query_file[:-4] != "negative") and (
```
Comment (msm-code) [inline e2e-correctness-tests/test_ursa_query.py:49]: `not query_file.startswith('negative')`

File: e2e-correctness-tests/test_ursa_query.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                (len(resp) != 1) or (query_file[:-4] not in resp[0])
```
Comment (msm-code) [inline e2e-correctness-tests/test_ursa_query.py:50]: why `!= 1`? multiple results are OK I think, but maybe I don't understand this check.

Comment (dskwhitehat) [reply e2e-correctness-tests/test_ursa_query.py:50]: I assumed that there will be only one file for corresponding yara rule. Then the test is simple. Check if there is only one corresponding file found. There could be also similar file that will be wrongly classified as matching the rule. What then? I could also store somewhere the number of files that should be found? Another issue is that the number can be correct but completely different wrongly matched files could be found. Maybe the probability of this is low but still exists... It's up to you what could be done...

Comment (dskwhitehat) [reply e2e-correctness-tests/test_ursa_query.py:50]: Another idea is to map beforehand (e.g. using python dictionary) yara rules to files that match the rule. In order to achieve this, I should know what those files are...

**Comment by msm-code** [comment]
I don't actually remember why this PR was never merged - looks complete or almost complete.

Will take a look in a few days.

**Comment by msm-code** [comment]
This was already incorporated as a part of mquery.

**Pull request closed without merging**